scripts/send-schedules.gmp

```diff
@@ -71,45 +67,21 @@ def parse_send_xml_tree(gmp, xml_tree):
         if comment is None:
             comment = ''
 
-        first_time = {}
-        f_time = schedule.find('first_time').text
-        f_time = dateutil.parser.parse(f_time)
-        first_time['day_of_month'] = f_time.day
-        first_time['hour'] = f_time.hour
-        first_time['minute'] = f_time.minute
-        first_time['month'] = f_time.month
-        first_time['year'] = f_time.year
-
-        duration = schedule.find('simple_duration').text
-        duration_unit = schedule.find('simple_duration').find('unit').text
-
-        period = schedule.find('simple_period').text
-        period_unit = schedule.find('simple_period').find('unit').text
+        ical = schedule.find('icalendar').text
 
         timezone_abbrev = schedule.find('timezone_abbrev').text
 
         gmp.create_schedule(
-            name=name,
-            first_time_minute=f_time.minute,
-            first_time_hour=f_time.hour,
-            first_time_day_of_month=f_time.day,
-            first_time_month=f_time.month,
-            first_time_year=f_time.year,
-            duration=duration,
-            duration_unit=duration_unit,
-            period=period,
-            period_unit=period_unit,
-            comment=comment,
-            timezone=timezone_abbrev,
+            name=name, comment=comment, timezone=timezone_abbrev, icalendar=ical
         )
 
 
-def main():
+def main(gmp, args):
     # pylint: disable=undefined-variable
 
     check_args(args)
 
```

Comment:
Could you add a check for using gmp 8 here e.g.
```python
if int(gmp.get_protocol_version()[0]) < 8:
    print("This script requires GMP version 8")
    sys.exit(1)
```

This lets me think about returning the version in get_protocol_version() as a tuple instead of a string...

Reply:
Yes this is a good idea as it will not work with gmp 7 anymore.